# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1326068635 18000
#      Sun Jan 08 19:23:55 2012 -0500
# Branch stable
# Node ID 447b57ebc27b6b6be2bcaf1f17871a13f438d1e2
# Parent  47a5e29ecc0b0250562629874a2f4ff006fe8637
Validate suffix when printing to {eps,ps,pdf}latex devices.

* __fltk_print__.m: gl2ps is not agnostic to file suffix.
* __gnuplot_print__.m: gnuplot is not agnostic to file suffix.

diff --git a/scripts/plot/private/__fltk_print__.m b/scripts/plot/private/__fltk_print__.m
--- a/scripts/plot/private/__fltk_print__.m
+++ b/scripts/plot/private/__fltk_print__.m
@@ -47,25 +47,27 @@ function opts = __fltk_print__ (opts)
   case {"epslatex", "pslatex", "pdflatex", "epslatexstandalone", ...
         "pslatexstandalone", "pdflatexstandalone"}
     ## format GL2PS_TEX
     n = find (opts.devopt == "l", 1);
     suffix = opts.devopt(1:n-1);
     dot = find (opts.name == ".", 1, "last");
     if ((! isempty (dot))
         && any (strcmpi (opts.name(dot:end), ...
-                {".eps", ".ps", ".pdf", ".tex", "."})))
+                {strcat(".", suffix), ".tex", "."})))
       name = opts.name(1:dot-1);
       if (dot < numel (opts.name)
           && any (strcmpi (opts.name(dot+1:end), {"eps", "ps", "pdf"})))
         ## If user provides eps/ps/pdf suffix, use it.
         suffix = opts.name(dot+1:end);
       endif
-    elseif (dot == numel (opts.name))
-      name = opts.name;
+    else
+      error ("print:invalid-suffix", 
+             "invalid suffix `%s' for device `%s'.",
+             opts.name(dot:end), lower (opts.devopt));
     endif
     gl2ps_device = {sprintf("%snotxt", lower (suffix))};
     gl2ps_device{2} = "tex";
     if (dos_shell)
       ## FIXME - this will only work on MinGW with the MSYS shell
       pipeline = {sprintf("cat > %s-inc.%s", name, suffix)};
       pipeline{2} = sprintf ("cat > %s.tex", name);
     else
diff --git a/scripts/plot/private/__gnuplot_print__.m b/scripts/plot/private/__gnuplot_print__.m
--- a/scripts/plot/private/__gnuplot_print__.m
+++ b/scripts/plot/private/__gnuplot_print__.m
@@ -64,21 +64,24 @@ function opts = __gnuplot_print__ (opts)
       pipeline = {sprintf("%s %s",
                           opts.epstool_cmd (opts, tmp_file, opts.name),
                           cleanup)};
     else
       eps_drawnow (opts, opts.name, gp_opts);
     endif
   case {"epslatex", "pslatex", "pstex", "epslatexstandalone"}
     dot = find (opts.name == ".", 1, "last");
+    n = find (opts.devopt == "l", 1);
+    suffix = opts.devopt(1:n-1);
     if (! isempty (dot))
-      if (any (strcmpi (opts.name(dot:end), {".eps", ".ps", ".pdf", ".tex", "."})))
+      if (any (strcmpi (opts.name(dot:end), {strcat(".", suffix), ".tex", "."})))
         name = opts.name(1:dot-1);
       else
-        error ("print:invalid-suffix", "unrecognized file name suffix `%s' for %s output type",
+        error ("print:invalid-suffix", 
+               "invalid suffix `%s' for device `%s'.",
                opts.name(dot:end), lower (opts.devopt));
       endif
     endif
     if (strfind (opts.devopt, "standalone"))
       term = sprintf ("%s ",
                       strrep (opts.devopt, "standalone", " standalone"));
     else
       term = sprintf ("%s ", opts.devopt);
